File: _pages/code/01/java-cpp/Dice.java
import java.util.Random;

class Dice {
    int sides;
    Random r = new Random();

    public Dice() {
        this.sides = 6;
    }

    public Dice(int sides) {
        this.sides = sides;
    }

    public int roll() {
        return r.nextInt(sides)+1;
    }

    public static void main(String[] args) {
        Dice die = new Dice(12);
        int result = die.roll();
        System.out.println(result);
    }
}

File: _pages/code/01/java-cpp/dice.cpp
#include <cstdlib>
#include <iostream>

class Dice {

    int sides;

    public: 
        Dice() {
            sides = 6;
        }
        Dice(int s) {
            sides = s;
        }
        
        ~Dice(){}

        int roll() {
            return (std::rand() % sides) + 1;
        }

};

int main() {
    Dice die(12);
    int result = die.roll();
    std::cout << result << std::endl;
}